logstash-core/src/test/java/org/logstash/common/io/DeadLetterQueueWriterTest.java

```diff
@@ -92,4 +98,33 @@ public void testWrite() throws Exception {
         writer.writeEntry(entry);
         writer.close();
     }
+
+    @Test
+    public void testDoesNotWriteBeyondLimit() throws Exception {
+        DLQEntry entry = new DLQEntry(new Event(), "type", "id", "reason");
+
+        int payloadLength = RECORD_HEADER_SIZE + VERSION_SIZE + entry.serialize().length;
+        long queueLength = payloadLength * 5;
+        DeadLetterQueueWriter writer = new DeadLetterQueueWriter(dir, payloadLength, queueLength);
```

Comment:
Actually, I can't - DeadLetterQueueWriter doesn't implement Closeable or AutoCloseable. Support for that is coming in a refactor I am working on. I'll update this with a try... finally to split the difference ;)